### PR TITLE
GCP Logger Refactor

### LINKED ISSUE #16738: Refactor GCP module to use Logger tool and log relevant information
| Affected integration |
|---|
| GCloud|

## Description

As described in #16717, To standardize the logging a refactor in this cloud module needs to be done to integrate the logging tool developed in the mentioned issue.

Logging refactor on the module should be done to log relevant information about the process.

Unit tests should be reviewed as well to adapt to the new logging method.

## Tasks
- [x] Test in a manager.
- [x] Test in an agent.
- [x] Unit tests without failures. Updated if there are any relevant changes.
- [x] Integration tests without failures. Updated if there are any relevant changes.
- [x] Update the documentation if necessary.
- [x] Add entry to the changelog if necessary.

### PR DESCRIPTION
**Pull request by EduLeon12**
|Related issue|
|---|
|Closes #16738 |

## Description

As part of #16717, This PR tends to update the GCP module to the developed logger and parser,   

## Tests

All related test results can be found [here](https://github.com/wazuh/wazuh/issues/16738#issuecomment-1745564104). 

```
(unit-test) eduardoleon@pop-os:~/git/wazuh/wodles/gcloud(16738-refactor-gcp-logs)$ pytest --disable-warnings tests/
==================================== test session starts ====================================
platform linux -- Python 3.9.16, pytest-7.4.2, pluggy-0.13.1
rootdir: /home/eduardoleon/git/wazuh/wodles/gcloud
plugins: anyio-3.6.2, aiohttp-1.0.4, trio-0.7.0, html-3.0.0, asyncio-0.18.1, cov-3.0.0, metadata-2.0.2
asyncio: mode=auto
collected 73 items

tests/test_bucket.py .................................                                [ 45%]
tests/test_gcloud.py .........                                                        [ 57%]
tests/test_integration.py ........                                                    [ 68%]
tests/test_subscriber.py ..............                                               [ 87%]
tests/test_tools.py .........                                                         [100%]

=============================== 73 passed, 1 warning in 0.42s ===============================
```

### Coverage report

```
========================= 73 passed, 1 warning in 0.55s =========================
Name                                      Stmts   Miss  Cover   Missing
-----------------------------------------------------------------------
wodles/gcloud/buckets/access_logs.py         17      0   100%
wodles/gcloud/buckets/bucket.py             128      2    98%   28-29
wodles/gcloud/exceptions.py                  23      0   100%
wodles/gcloud/gcloud.py                      70      1    99%   134
wodles/gcloud/integration.py                 39      2    95%   40, 101
wodles/gcloud/pubsub/subscriber.py           61      2    97%   24-25
wodles/gcloud/tests/test_bucket.py          228      0   100%
wodles/gcloud/tests/test_gcloud.py           49      0   100%
wodles/gcloud/tests/test_integration.py      51      0   100%
wodles/gcloud/tests/test_subscriber.py       74      0   100%
wodles/gcloud/tests/test_tools.py            26      0   100%
wodles/gcloud/tools.py                       27      0   100%
wodles/shared/__init__.py                     0      0   100%
wodles/shared/wazuh_cloud_logger.py          29      0   100%
wodles/utils.py                              57     27    53%   29-30, 39-40, 59-66, 84-98, 111, 124, 137
-----------------------------------------------------------------------
TOTAL                                       879     34    96%

```

### CODE REVIEW

File: wodles/gcloud/gcloud.py [hunk 1]
```diff
@@ -77,33 +96,32 @@ def main():
...
-        logging_func = logger.critical if \
-            isinstance(gcloud_exception, exceptions.WazuhIntegrationInternalError) else \
-            logger.error
-
-        logging_func(f'An exception happened while running the wodle: {gcloud_exception}', exc_info=log_level == 1)
+        gcp_logger.error(f'An exception happened while running the wodle: {gcloud_exception}')
         exit(gcloud_exception.errcode)
 
     except Exception as e:
-        logger.critical(f'Unknown error: {e}', exc_info=True)
```
Comment (fdalmaup) [inline wodles/gcloud/gcloud.py:101]: The `exc_info` parameter defines if additional information regarding the exception should be included. We should analyze if this should be added for the `WazuhCloudLogger.error` method.

File: wodles/gcloud/pubsub/subscriber.py [hunk 2]
```diff
@@ -5,27 +5,29 @@
 # Created by Wazuh, Inc. <info@wazuh.com>.
 # This program is free software; you can redistribute
 # it and/or modify it under the terms of GPLv2
-import logging
+
+"""This module contains the class for sending events from Google Cloud subscriber service to Wazuh. """
```
Comment (fdalmaup) [inline wodles/gcloud/pubsub/subscriber.py:9]:
```suggestion
"""This module contains the class for sending events from Google Cloud subscriber service to Wazuh."""
```

File: wodles/gcloud/tests/test_tools.py [hunk 3]
```diff
@@ -28,28 +41,21 @@ def test_get_script_arguments(capsys):
...
-@patch('tools.logging.getLogger')
-def test_get_stdout_logger(mock_logger, level, expected_logging_level):
-    """Test the get_stdout_logger function created a logger object with the expected logging level value."""
-    with patch('tools.logger') as mock_l:
-        mock_l.setLevel = MagicMock()
-        tools.get_stdout_logger(name='test', level=level)
-        mock_l.setLevel.assert_called_with(expected_logging_level)
-
-
-def test_arg_valid_date():
```
Comment (fdalmaup) [inline wodles/gcloud/tests/test_tools.py:52]: Why was this test removed?

File: wodles/gcloud/tests/test_tools.py [hunk 4]
```diff
@@ -28,28 +41,21 @@ def test_get_script_arguments(capsys):
...
-    """Test arg_valid_dates raises an error when a date parameter doesn't have a valid format."""
-    with pytest.raises(argparse.ArgumentTypeError):
-        tools.arg_valid_date('invalid_date')
+def test_log_methods(gcp_logger, caplog, log_level, log_method, expected_message):
```
Comment (fdalmaup) [inline wodles/gcloud/tests/test_tools.py:57]: The logging is no longer handled in the `tools.py` file unlike it was with the `get_stdout_logger` function, this test could be removed.

**Review by fdalmaup — CHANGES_REQUESTED** [review]

File: wodles/gcloud/buckets/access_logs.py [hunk 5]
```diff
@@ -33,7 +36,7 @@ def __init__(self, credentials_file: str, logger: logging.Logger, **kwargs):
         self.db_table_name = "access_logs"
 
     def load_information_from_file(self, msg: str):
-        """Load the contents of an Access Logs blob and process them.
+        """Load the contents of an Access Logs bucket and process them.
```
[the remark this answers is not in the document]
Comment (EduLeon12) [reply wodles/gcloud/buckets/access_logs.py:39]: :pepe_notes:

File: wodles/gcloud/gcloud.py (hunk 1 above)
Comment (EduLeon12) [reply wodles/gcloud/gcloud.py:101]: Included

File: wodles/gcloud/tests/test_tools.py (hunk 4 above)
Comment (EduLeon12) [reply wodles/gcloud/tests/test_tools.py:57]: Moved

**Review by fdalmaup — APPROVED** [review]
LGTM!

**Review by Selutario — CHANGES_REQUESTED** [review]

**Review by Selutario — CHANGES_REQUESTED** [review]
LGTM, it seems there are some conflicts in the PR which need to be fixed before merging :eyes:

**Review by Selutario — APPROVED** [review]

**Pull request merged**